readium/navigator/src/main/java/org/readium/r2/navigator/media2/MediaNavigator.kt

```diff
@@ -0,0 +1,99 @@
+package org.readium.r2.navigator.media2
+
+import kotlinx.coroutines.flow.StateFlow
+import org.readium.r2.navigator.ExperimentalAudiobook
+import org.readium.r2.navigator.Navigator
+import org.readium.r2.shared.publication.Link
+import org.readium.r2.shared.publication.Locator
+import org.readium.r2.shared.util.Try
+import kotlin.time.Duration
+import kotlin.time.ExperimentalTime
+
+/**
+ * A navigator rendering an audio or video publication.
+ */
+@ExperimentalAudiobook
+@OptIn(ExperimentalTime::class)
+interface MediaNavigator : Navigator {
+
+    /**
+     * Indicates the navigator current state.
+     */
+    val playback: StateFlow<Playback>
+
+    /**
+     * Resumes or start the playback at the current location.
+     */
+    suspend fun play(): Try<Unit, Exception>
+
+    /**
+     * Pauses the playback.
+     */
+    suspend fun pause(): Try<Unit, Exception>
+
+    /**
+     * Seeks to the given time at the given resource.
+     */
+
+    suspend fun seek(index: Int, position: Duration): Try<Unit, Exception>
+
+    /**
+     * Skips to a little amount of time later.
+     */
+
+    suspend fun goForward(): Try<Unit, Exception>
+
+    /**
+     * Skips to a little amount of time before.
+     */
+
+    suspend fun goBackward(): Try<Unit, Exception>
+
+    /**
+     * Seeks to the beginning of the given link.
+     */
+
+    suspend fun go(link: Link): Try<Unit, Exception>
+
+    /**
+     * Seeks to the given locator.
+     */
+
+    suspend fun go(locator: Locator): Try<Unit, Exception>
```

Comment:
These are part of the `Navigator` interface already. I know they are less than ideal and not using coroutines, but this defeats the purpose of having a common interface (which is useful for shared screens, for example the table of contents or bookmarks). We can improve these APIs in Readium 3.x.

Reply:
NYPL doesn't use the common interface (so far at least), and I suppose we're not alone in doing so. It would be a pity to start a new navigator with technical debt, so I suggest we keep the matching methods public in  `MediaSessionNavigator` and remove them from the interface. Maybe that's what you were thinking about.

Reply:
Are you not using the common interface because you have your own EPUB navigator? 

I'm really surprised you wouldn't use the share interface if you used more than one Readium navigator, for example it's used by the test app here: https://github.com/readium/kotlin-toolkit/blob/e2eeb4aea86d4d20726ebc71a33872a6c0bdb219/test-app/src/main/java/org/readium/r2/testapp/reader/BaseReaderFragment.kt#L115-L116

Reply:
But yeah, I'm fine if you remove the suspending versions from the interface but keep them in the implementation, for now.

Reply:
We have one custom `Viewer` interface which is used by the `ServiceLoader` to dynamically load all available `Viewer` on the class path. This enables us to have a different set of viewers in different apps. Thus, we used to have each viewer in a separate module.
I think the benefits of the use of the common interface with both visual and audio navigators is quite tight. It's likely to save a couple of lines of code.